ui/app/AppLayouts/CommunitiesPortal/popups/CreateCommunityPopup.qml

```diff
@@ -58,7 +59,278 @@ StatusStackModal {
         }
     }
 
-    onAboutToShow: nameInput.input.edit.forceActiveFocus()
+    readonly property var clearFilesButton: StatusButton {
+        font.weight: Font.Medium
+        text: qsTr("Clear all")
+        type: StatusBaseButton.Type.Danger
+        visible: root.currentItem.objectName === "discordFileListView" // no better way to address the current item in the stack :/
+        enabled: !fileListView.fileListModelEmpty && !root.store.discordDataExtractionInProgress
+        onClicked: root.store.clearFileList()
+    }
+
+    rightButtons: [clearFilesButton, nextButton, finishButton]
+
+    onAboutToShow: {
+        nameInput.input.edit.forceActiveFocus()
+        if (root.isDiscordImport) {
+            for (let i = 0; i < discordPages.length; i++) {
+                stackItems.push(discordPages[i])
+            }
+        }
+    }
+
+    onClosed: {
+        if (root.isDiscordImport) {
+            root.store.clearFileList()
+            root.store.clearDiscordCategoriesAndChannels()
+        }
+        destroy()
```

Comment:
@PascalPrecht calling destroy from within an object is possible, but that means this community popup should always be dynamically instantiated, giving that I am always for the option for a more general approach and let a caller to take care about destruction.

Reply:
Moved the destroy call to the call side of this popup